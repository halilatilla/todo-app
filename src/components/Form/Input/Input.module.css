.input {
  @apply mt-1 block w-full rounded-md border-gray-300  text-gray-900 shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50;
}
