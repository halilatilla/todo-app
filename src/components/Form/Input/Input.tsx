import { FC } from 'react'
import classnames from 'classnames'

import styles from './Input.module.css'

interface Props {
  label?: string
  type?: string
  placeholder?: string
  defaultValue?: string
  className?: string
  classNameInput?: string
  value?: string | number
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
}

const Input: FC<Props> = ({ label, type = 'text', className, classNameInput, ...rest }) => {
  return (
    <label className={classnames('block', className)}>
      {label && <span className="text-sm capitalize text-gray-300">{label}</span>}
      <input type={type} className={classnames(styles.input, classNameInput)} {...rest} />
    </label>
  )
}

export default Input
