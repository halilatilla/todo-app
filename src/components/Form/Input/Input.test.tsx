import { render, screen } from '@testing-library/react'

import Input from './Input'

describe('Input', () => {
  const DATA = {
    onChange: () => {},
  }

  it('renders correctly', () => {
    const { container } = render(<Input {...DATA} />)
    expect(container).toMatchSnapshot()
  })

  it('renders Input', () => {
    render(<Input {...DATA} />)

    const input = screen.getByRole('textbox')

    expect(input).toBeInTheDocument()
  })
})
