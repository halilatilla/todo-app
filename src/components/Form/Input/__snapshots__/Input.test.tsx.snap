// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div>
  <label
    class="block"
  >
    <input
      class="input"
      type="text"
    />
  </label>
</div>
`;
